controller/tests/array_action/svc/array_mediator_svc_test.py

```diff
@@ -341,32 +411,42 @@ def test_delete_snapshot_no_fcmap_id_raise_snapshot_not_found(self):
         with self.assertRaises(array_errors.ObjectNotFoundError):
             self.svc.delete_snapshot("test_snap")
 
-    def test_delete_snapshot_call_rmvolume(self):
-        self._prepare_mocks_for_get_snapshot()
-
-        self.svc.delete_snapshot("test_snap")
-
-        self.svc.client.svctask.rmvolume.assert_called_once_with(vdisk_id="test_snap")
-
     def test_delete_snapshot_call_rmfcmap(self):
-        self._prepare_mocks_for_get_snapshot()
-
+        self._prepare_mocks_for_delete_snapshot()
+        fcmaps_as_target = self.fcmaps
+        self.svc.client.svcinfo.lsfcmap.side_effect = [Mock(as_list=fcmaps_as_target), Mock(as_list=[])]
         self.svc.delete_snapshot("test_snap")
 
         self.svc.client.svctask.rmfcmap.assert_called_once_with(object_id="test_fc_id", force=True)
 
     @patch("controller.array_action.array_mediator_svc.is_warning_message")
     def test_delete_snapshot_rmvolume_error(self, mock_warning):
-        self._prepare_mocks_for_get_snapshot()
+        self._prepare_mocks_for_delete_snapshot()
         mock_warning.return_value = False
         self.svc.client.svctask.rmvolume.side_effect = [
             CLIFailureError("Failed")]
 
         with self.assertRaises(CLIFailureError):
             self.svc.delete_snapshot("test_snap")
 
+    def test_delete_snapshot_still_copy_fcmaps_not_removed(self):
+        self._prepare_mocks_for_object_still_in_use()
+        fcmaps_as_target = self.fcmaps
+        fcmaps_as_source = [Munch(
+            {'source_vdisk_name': 'test_snap',
+             'target_vdisk_name': 'target_name',
+             'id': 'test_fc_id',
+             'status': "not_good",
```

Comment:
why not just `fcmaps_as_source[0].status = "not good"` like in `test_delete_volume_still_copy_fcmaps_not_removed`?

Reply:
no resaon.. done

Reply:
I'm referring to the string value. I'm referring to the unnecessary entire-dict assignment.

Reply:
oh, because Im using the` self.fcmap` for target, if Im gonna use it with source too it will change the target too (reference and such..)
so its or this or deep copy for the `self.fcmaps` .....

Reply:
can't we just put two separate fcmaps in `self` during setUp, and each test could choose which to use?

Reply:
well, we can, but to change now a lot of tests?

Reply:
I thought it would mainly require changes in the `_prepare` methods.
if it's too much change, feel free to open a ticket and write its number here instead.
we don't want the svc UTs to look like ds8k UTs...

Reply:
can you plz open a ticket?

Reply:
it's supposed to be part of this ticket.
if the ticket is too long, the assignee can always open follow up tickets, but opening follow ups is also part of the ticket.
it's not urgent. we can keep this conversation open until you get to it

Reply:
ok, I done it.